src/components/Tooltip/Tooltip.js

```diff
@@ -46,35 +96,82 @@ class Tooltip extends Component {
             : tooltip;
     }
 
-    handleFocus = () => this.setState({ showTooltip: true })
+    getLeftPosition = position => ({
+        [TooltipPosition.TOP_CENTER]:
+            Math.floor((this.state.wrapperSize.width - this.state.tooltipSize.width) / 2),
+        [TooltipPosition.BOTTOM_CENTER]:
+            Math.floor((this.state.wrapperSize.width - this.state.tooltipSize.width) / 2),
+        [TooltipPosition.BOTTOM_LEFT]: -(this.state.tooltipSize.width + this.props.tooltipOffset),
+        [TooltipPosition.LEFT]: -(this.state.tooltipSize.width + this.props.tooltipOffset),
+        [TooltipPosition.TOP_LEFT]: -(this.state.tooltipSize.width + this.props.tooltipOffset),
+        [TooltipPosition.TOP_RIGHT]: this.state.wrapperSize.width + this.props.tooltipOffset,
+        [TooltipPosition.RIGHT]: this.state.wrapperSize.width + this.props.tooltipOffset,
+        [TooltipPosition.BOTTOM_RIGHT]: this.state.wrapperSize.width + this.props.tooltipOffset,
+    })[position || TooltipPosition.BOTTOM_RIGHT];
+
+    getTopPosition = position => ({
+        [TooltipPosition.TOP_CENTER]: -(this.state.tooltipSize.height + this.props.tooltipOffset),
+        [TooltipPosition.RIGHT]:
+            Math.floor((this.state.wrapperSize.height - this.state.tooltipSize.height) / 2),
+        [TooltipPosition.LEFT]:
+            Math.floor((this.state.wrapperSize.height - this.state.tooltipSize.height) / 2),
+        [TooltipPosition.BOTTOM_RIGHT]: this.state.wrapperSize.height,
+        [TooltipPosition.BOTTOM_LEFT]: this.state.wrapperSize.height,
+        [TooltipPosition.BOTTOM_CENTER]: this.state.wrapperSize.height + this.props.tooltipOffset,
+        [TooltipPosition.TOP_RIGHT]: -(this.state.tooltipSize.height),
+        [TooltipPosition.TOP_LEFT]: -(this.state.tooltipSize.height),
+    })[position || TooltipPosition.TOP_LEFT];
+
+    handleFocus = e =>
+        this.setState({ showTooltip: true, autoPosition: positionFromFocusEvent(e) });
```

Comment:
I know this is spread out in our codebase, but I'm not a fan of using arrow functions as class properties.
https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1
What's your view on this?

Reply:
Interesting. I hadn't seen that article before. I have no strong opinion on either way of declaring class properties tbh. I'm happy to change this to a function declaration.

I not too worried about any of the prototype/inheritance concerns from the article as we don't really do too much OO JS. I guess if there's a potential performance benefit we may as well take it though